python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/asset_decorator.py

```diff
@@ -14,13 +14,57 @@
 from .translator import DagsterDltTranslator
 
 
+def build_dlt_asset_specs(
+    dlt_source: DltSource,
+    dlt_pipeline: Pipeline,
+    dagster_dlt_translator: Optional[DagsterDltTranslator] = None,
+) -> List[AssetSpec]:
+    """Build a list of asset specs from a dlt source and pipeline.
+
+    Args:
+        dlt_source (DltSource): dlt source object
+        dlt_pipeline (Pipeline): dlt pipeline object
+        dagster_dlt_translator (Optional[DagsterDltTranslator]): Allows customizing how to
+            map dlt project to asset keys and asset metadata.
+
+    Returns:
+        List[AssetSpec] list of asset specs from dlt source and pipeline
+
+    """
+    dagster_dlt_translator = dagster_dlt_translator or DagsterDltTranslator()
+    destination_type = dlt_pipeline.destination.destination_name
+    return [
+        AssetSpec(
+            key=dagster_dlt_translator.get_asset_key(dlt_source_resource),
+            auto_materialize_policy=dagster_dlt_translator.get_auto_materialize_policy(
+                dlt_source_resource
+            ),
+            deps=dagster_dlt_translator.get_deps_asset_keys(dlt_source_resource),
+            description=dagster_dlt_translator.get_description(dlt_source_resource),
+            group_name=dagster_dlt_translator.get_group_name(dlt_source_resource),
+            metadata={
+                META_KEY_SOURCE: dlt_source,
+                META_KEY_PIPELINE: dlt_pipeline,
+                META_KEY_TRANSLATOR: dagster_dlt_translator,
+                **dagster_dlt_translator.get_metadata(dlt_source_resource),
+            },
+            owners=dagster_dlt_translator.get_owners(dlt_source_resource),
+            tags={
+                "dagster/storage_kind": destination_type,
+                **dagster_dlt_translator.get_tags(dlt_source_resource),
+            },
+        )
+        for dlt_source_resource in dlt_source.selected_resources.values()
+    ]
+
+
 def dlt_assets(
     *,
     dlt_source: DltSource,
     dlt_pipeline: Pipeline,
     name: Optional[str] = None,
     group_name: Optional[str] = None,
-    dlt_dagster_translator: Optional[DagsterDltTranslator] = None,
+    dagster_dlt_translator: Optional[DagsterDltTranslator] = None,
```

Comment:
This is technically a breaking change, so we should document this in the `MIGRATION.md` for `dagster==1.8`.

Reply:
Good call, I may revert this change in that case.

Unless you think this level of consistency would be beneficial, as 1.8 is an appropriate time for breaking changes.

Reply:
I've used `deprecate_params` and made both supported for the time being. Does that seem ergonomic?

https://github.com/dagster-io/dagster/pull/23187/commits/ee7489111d9ac776a097c28ee6f42c0c62e59c84